### PR TITLE
Ethereum Personal Management API

### PR DESCRIPTION
**Pull request by beatrizsanchez**
Added support for Ethereum's [Personal API](https://github.com/ethereum/go-ethereum/wiki/Management-APIs#personal)

### CODE REVIEW

File: src/main/java/org/web3j/protocol/core/personal/Personal.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.web3j.protocol.core.personal.methods.response.EcRecovered;
+import org.web3j.protocol.core.personal.methods.response.ImportedRawKey;
+import org.web3j.protocol.core.personal.methods.response.LockedAccount;
+import org.web3j.protocol.core.personal.methods.response.NewAccount;
+import org.web3j.protocol.core.personal.methods.response.Signed;
+import org.web3j.protocol.core.personal.methods.response.Transaction;
+import org.web3j.protocol.core.personal.methods.response.UnlockedAccount;
+import org.web3j.protocol.parity.methods.response.PersonalListAccounts;
+
+public interface Personal{
```
Comment (conor10) [inline src/main/java/org/web3j/protocol/core/personal/Personal.java]: As this is the Geth specific list of Personal API calls, I suggest you split this interface in two:

- Those calls that are common to both Parity & Geth can be in org.web3j.protocol.admin.Personal
- Those calls that are specific to Geth can be in org.web3j.protocol.geth.Geth

File: src/main/java/org/web3j/protocol/core/personal/JsonRpc2_ExtendedWeb3j.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+implements Personal {
+
+	public JsonRpc2_ExtendedWeb3j(Web3jService web3jService) {
+		super(web3jService);
+	}
+
+	@Override
+	public Request<?, ImportedRawKey> personalImportRawKey(String hexStringNoPrefix, String passphrase){
+		return new Request<>(	
+				"personal_importRawKey",
```
Comment (conor10) [inline src/main/java/org/web3j/protocol/core/personal/JsonRpc2_ExtendedWeb3j.java]: Indentation should be 8 characters under the return statement, not 16. Please ensure this is consistent in all files in the PR.

File: src/main/java/org/web3j/protocol/core/personal/JsonRpc2_ExtendedWeb3j.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import org.web3j.protocol.core.personal.methods.response.EcRecovered;
+import org.web3j.protocol.core.personal.methods.response.ImportedRawKey;
+import org.web3j.protocol.core.personal.methods.response.LockedAccount;
+import org.web3j.protocol.core.personal.methods.response.NewAccount;
+import org.web3j.protocol.core.personal.methods.response.Signed;
+import org.web3j.protocol.core.personal.methods.response.Transaction;
+import org.web3j.protocol.core.personal.methods.response.UnlockedAccount;
+import org.web3j.protocol.parity.methods.response.PersonalListAccounts;
+
+public class JsonRpc2_ExtendedWeb3j extends JsonRpc2_0Web3j
```
Comment (conor10) [inline src/main/java/org/web3j/protocol/core/personal/JsonRpc2_ExtendedWeb3j.java]: Once the Interface has been split (see comment below). I recommend you name the implementation class files:

- org.web3j.protocol.admin.JsonRpc2_0Admin
- org.web3j.protocol.geth.JsonRpc2_0Geth

File: src/main/java/org/web3j/protocol/core/personal/JsonRpc2_ExtendedWeb3j.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import org.web3j.protocol.core.Request;
+import org.web3j.protocol.core.personal.methods.response.EcRecovered;
+import org.web3j.protocol.core.personal.methods.response.ImportedRawKey;
+import org.web3j.protocol.core.personal.methods.response.LockedAccount;
+import org.web3j.protocol.core.personal.methods.response.NewAccount;
+import org.web3j.protocol.core.personal.methods.response.Signed;
+import org.web3j.protocol.core.personal.methods.response.Transaction;
+import org.web3j.protocol.core.personal.methods.response.UnlockedAccount;
+import org.web3j.protocol.parity.methods.response.PersonalListAccounts;
+
```
Comment (conor10) [inline src/main/java/org/web3j/protocol/core/personal/JsonRpc2_ExtendedWeb3j.java]: Request & response tests will need to be included for all added methods. See the existing Parity tests for examples at https://github.com/web3j/web3j/tree/master/src/test/java/org/web3j/protocol/parity.

File: src/main/java/org/web3j/protocol/core/personal/Personal.java (hunk 1 above)
Comment (beatrizsanchez) [inline src/main/java/org/web3j/protocol/core/personal/Personal.java]: > Those calls that are common to both Parity & Geth can be in org.web3j.protocol.admin.Personal

do you mean that parity is to be moved into that package as well? wouldn't it be better that  they all were located in org.web3j.protocol.geth.Geth?

Comment (conor10) [inline src/main/java/org/web3j/protocol/core/personal/Personal.java]: My thinking is, those that are already in org.web3j.protocol.parity.Parity, but are common to Geth too, should be moved into a new Personal interface, i.e.:

- personal_listAccounts
- personal_newAccount
- personal_sendTransaction
- personal_unlockAccount

That way there's no duplication of code between the Parity & Geth specific client support in web3j.

The remainder that are exclusive to Parity should stay in org.web3j.protocol.parity.Parity. Then those that you've added which are exclusive to Geth should be in org.web3j.protocol.geth.Geth.

**Comment by conor10** [comment]
Closing as completed by https://github.com/web3j/web3j/pull/147.
@beatrizsanchez thanks for getting this started.

**Pull request closed without merging**